pkg/pods/pod.go

```diff
@@ -593,6 +596,34 @@ func (pod *Pod) Launchables(manifest Manifest) ([]launch.Launchable, error) {
 	return launchables, nil
 }
 
+func (pod *Pod) SetFinishExec(finishExec []string) {
+	var runAs string
+
+	if manifest, err := pod.CurrentManifest(); err == nil {
```

Comment:
Good catch. This is definitely strange - if the `run_as` were to change between deploys, we don't want to be caught with the _last_ deploy's user. Most functions to `Pod` also pass the manifest we're operating on, why not pass it here too?